File: program92.cpp
#include<iostream>

using namespace std;

int SumFactors(int iNo)
{
int i=0;
int sum=0;
for(i=0;i<=iNo;i++)
{
    if((iNo%i==0))
    {
        
        sum=sum+i;
    }
}
return sum;
}

int main()
{
    int iValue=0;
    int iRet=0;
    cout<<"enter number:"<<endl;
    cin>>iValue;
    iRet=SumFactors(iValue);

    cout<<"summation of factors is:"<<iRet<<endl;

    return 0;
}

File: program92.java
import java.util.*;


public class program92
 {
    public static void main(String Arg[])
    {
        Scanner sobj =new Scanner(System.in);

        int iValue=0;
        int i=0;
        int sum=0;

        System.out.println("enter number: ");

        iValue=sobj.nextInt();

        for(i=1;i<=(iValue/2);i++)
        {
            if((iValue%i)==0)
            {
                sum=sum+i;
            }
        }
        System.out.println("summ is "+sum);
    }
}
